File: III-OITF/p5/p5.py
import sys

def cambio_base(n, b):
    if n == 0:
        return "0"
    r = ""
    while n > 0:
        r = str(n % b) + r
        n //= b
    return r

for line in sys.stdin:
    nums = line.strip().split()

    if len(nums) == 2:
        if nums[0].isdigit() and nums[1].isdigit() and int(nums[1]) > 1 and int(nums[1]) < 11 and int(nums[0]) > 0:
            print(cambio_base(int(nums[0]), int(nums[1])))
        else:
            print("ERROR")
    else:
        print("ERROR")

File: III-OITF/p5/p5.cpp
#include <iostream>
#include <sstream>

void cambio_base(int n, int b) {
    if (n == 0) {
        std::cout << "0" << std::endl;
        return;
    }

    int resultado[32];
    int indice = 0;

    while (n > 0) {
        resultado[indice++] = n % b;
        n /= b;                      
    }

    for (int i = indice - 1; i >= 0; --i) {
        std::cout << resultado[i];
    }
    std::cout << std::endl;
}

int main() {
    std::string line;

    while (std::getline(std::cin, line)) {
        int n = 0, b = 0;
        bool error = false;

        std::stringstream ss(line);
        if (!(ss >> n >> b)) {
            error = true;
        }

        if (!error && b > 1 && b < 11 && n > 0) {
            cambio_base(n, b);
        } else {
            std::cout << "ERROR" << std::endl;
        }
    }
    return 0;
}